src/crimson/mon/MonClient.cc

```diff
@@ -1024,14 +1036,32 @@ Client::run_command(const std::vector<std::string>& cmd,
   m->cmd = cmd;
   m->set_data(bl);
   auto& req = mon_commands[tid];
-  return active_con->get_conn()->send(m).then([&req] {
+  return send_message(m).then([&req] {
     return req.get_future();
   });
 }
 
 seastar::future<> Client::send_message(MessageRef m)
 {
-  return active_con->get_conn()->send(m);
+  if (active_con) {
+    if (pending_messages.size()) {
+      send_pendings();
+    }
+    return active_con->get_conn()->send(m);
+  }
+  pending_messages.emplace_back(m);
+  return pending_messages.back().pr.get_future();;
+}
+
+void Client::send_pendings()
+{
+  if (active_con) {
```

Comment:
i think we can assume that we always have an `active_con` here.

Reply:
Um...I think there maybe a tiny chance that, between the running of this send_pendings() method and session reopening, the connection gets reset again, is this right?